actions.js

```diff
@@ -0,0 +1,56 @@
+
+/* eslint-disable import/prefer-default-export */
+import { Toast } from 'native-base';
+import CONFIG from './config.js';
+
+const API_URL = CONFIG.ENV === 'PROD' ? CONFIG.API_URL : CONFIG.TEST_API_URL;
+
+export function constructContactItemsWithToken(token) {
+  return async (dispatch) => {
+    try {
+      dispatch({
+        type: 'SET_USER_DATA',
+        data: {
+          refreshingContacts: true
+        }
+      });
+      const response = await fetch(`${API_URL}/contacts/${token._id}?token=${token.api_key}`);
+      if (!response.ok) {
+        throw new Error(`Failed with status code: ${response.status}`);
+      }
+      const contactData = (await response.json()).contacts.map(contact => ({
+        name: `${contact.person.fname} ${contact.person.lname}`,
+        school: contact.school.name,
+        grade: contact.school.grade,
+        id: contact._id,
+        facebook_id: contact.facebook_id,
+        fullContact: contact
+      }));
+      if (contactData.length === 0) {
+        Toast.show({
+          text: 'Hmm, nobody\'s here, get in touch with Hey Mentor to get paired with someone.',
+          buttonText: 'Okay',
+          duration: 10000
+        });
+      }
+      dispatch({
+        type: 'SET_USER_DATA',
+        data: {
+          contactItem: contactData
+        }
+      });
+    } catch (err) {
+      Toast.show({
```

Comment:
Since we're now updating local Redux state, we should adjust how we handle and display these errors. 

If we fail to retrieve data from the backend, but we already have data in redux locally, we shouldn't show an error when the user loads the contacts list page. 

However, if the user force refreshes the contacts list (by scrolling up to refresh), we should show an error.

Reply:
we can add a boolean condition like `isInitialLoad` to handle it differently but it may be an overkill

Reply:
If the action itself doesn't handle the error logic, but propagates the errors up to the layer above, then the container component will know what to do with the error (display it or not, based on the container knowledge, such as what triggered the dispatch).

Reply:
fixed